### PR TITLE
Week 2 assessment yanxu2021

### PR DESCRIPTION
**Pull request by yanxu2021**
Please review. Appreciate and have a nice day~   Yan

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -18,48 +18,104 @@
...
+        expect(divideByThree("")).toEqual(`Please enter a number`)
+    })
+})
 // b) Create the function that makes the test pass.
-
+// If the number can be evenly divisible by 3, this number should % 3 === 0.
+// Using "if else" statement 
+const divideByThree = (num) =>{
+    if(typeof num !== `number`){
+       return `Please enter a number`
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:41]: Very good!

File: code-challenges.test.js [hunk 2]
```diff
@@ -18,48 +18,104 @@
...
+    //.map() is very useful high-order function, it saves lot of time to loop through array.
+    //.map(value, index, array)
+    let eachItem = arr.map(value =>{
+        return value[0].toUpperCase() + value.substring(1)
+    })  
+    //Only need value here, which is the value at each index of an array.
+    
+    //return eachItem.join(" ")//Output:Temperature Database Chopsticks Mango Deduction
+    return eachItem
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:82]: Readable! I like it

File: code-challenges.test.js [hunk 3]
```diff
@@ -18,48 +18,104 @@
...
 // b) Create the function that makes the test pass.
+// Loop though each letter in the string 
+// condition will be true:  ||  ----> log out the index
+const vowel = (str) =>{
+    for(let i = 0;i < str.length;i++){//As string's property, str.length, not str.length()
+        if(str[i] == "a" || str[i] == "e" || str[i] == "i" || str[i] == "o" || str[i] == "u" ){
+            return i
+        }
+    } 
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:112]: Good Work!

File: coding-concepts.js [hunk 4]
```diff
@@ -8,40 +8,59 @@
...
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer:[12, 14]
+// b) Verify and explain: As expected, the output: [ 12, 14 ]
+// The .filter() method creates an array filled with all array elements that pass a test(provided by a function). 
+// The .filter() method does not execute the function for empty array elements.
+// The .filter() method does not change the original array. 
+// value % 2 ===0, only picked/filtered even numbers.
+// To fix this output,
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:60]: Great Explanation

File: coding-concepts.js [hunk 5]
```diff
@@ -51,9 +70,19 @@ class Learn {
...
-// console.log(learnStudent)
-
-// a) Your answer:
-// b) Verify and explain:
+console.log(learnStudent)
+
+// a) Your answer:Learn { student: 'George', cohort: 'Charlie', year: 2021 }
+// b) Verify and explain: output as expected: Learn { student: 'George', cohort: 'Charlie', year: 2021 }
+// The output will be all attributes(data) because of console.log(learnStudent). Here use "new" to create an instance of "Learn".  
+// Try different output to see how it works.
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:84]: Amazing!

**Review by WalkerTexas-coder — APPROVED** [review]
Yan, everything about this is amazing! Your interview answers are well thought out and researched, Your concepts go the extra mile and your code logic is clean and simple. Great Work thank you for putting so much effort into this! Your PR is approved!

File: coding-concepts.js (hunk 5 above)
Comment (yanxu2021) [reply coding-concepts.js:84]: Thank you for your all comments and hard work.

**Pull request merged**